File: client/lib/models/user_model.dart
class User {
  final String? uid;
  final String? email;
  final String? password;
  final String? role;
  final String? fullname;
  final String? contactNo;
  final String? cnic;
  final String? profilePic;

  User({
    this.uid = '',
    this.email,
    this.password,
    this.role,
    this.fullname,
    this.contactNo,
    this.cnic,
    this.profilePic = '',
  });

  Map<String, dynamic> toJson() {
    return {
      'uid': '',
      'email': email,
      'password': password,
      'role': role,
      'fullname': fullname,
      'contact_no': contactNo,
      'cnic': cnic,
      'profilePic': '',
    };
  }

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      uid: json['_id'],
      email: json['email'],
      password: json['password'],
      role: json['role'],
      fullname: json['fullname'],
      contactNo: json['contact_no'],
      cnic: json['cnic'],
      profilePic: json['profilePic'],
    );
  }
}

File: server/models/userModel.js
const mongoose = require('mongoose');

const userSchema = new mongoose.Schema({
    email: {
        type: String,
        required: [true, "Please add email address"],
        unique: [true, "Email address already taken"],
    },
    password: {
        type: String,
        required: [true, "Please add password"],
        minlength: 8
    },
    role: {
        type: String,
        enum: ['guest', 'owner'],
        required: true
    },
    fullname: {
        type: String,
        required: true
    },
    contact_no: {
        type: String,
        required: true
    },
    cnic: {
        type: String,
        required: true,
        unique: true
    },
    profilePic: {
        type: String
    },
    createdAt: {
        type: Date,
        default: Date.now
    },
});

module.exports = mongoose.model('User', userSchema);
